### PR TITLE
Improved time scheme and their selection logic

### PR DESCRIPTION
**Pull request by timofeymukha**
* Adds docstring and a test suite to the `ext_time_scheme_t` type.
* Removes the coupling between EXT and BDF schemes, assuming they are always the same order. This particularly leads to the removal of the modified 3rd order EXT, which is meant to work with 2nd order BDF. My impression is that it makes no sense to have the mixed schemes, since we set one time order for the simulation. It's perhaps best that @pschlatt1 also takes a look to confirm my logic.

### CODE REVIEW

File: configure.ac [hunk 1]
```diff
@@ -299,10 +299,11 @@ AC_CONFIG_FILES([tests/stack/Makefile\
...
 		 tests/octree/Makefile\
-                 tests/vector/Makefile\
-                 tests/scratch_registry/Makefile\
-                 tests/fast3d/Makefile\
-                 tests/ext_bdf_scheme/Makefile])
+       tests/vector/Makefile\
+       tests/scratch_registry/Makefile\
+       tests/fast3d/Makefile\
+       tests/time_scheme/Makefile\
+       tests/time_scheme_controller/Makefile])
```
Comment (njansson) [inline configure.ac:306]:
```suggestion
                 tests/vector/Makefile\
                 tests/scratch_registry/Makefile\
                 tests/fast3d/Makefile\
                 tests/time_scheme/Makefile\
                 tests/time_scheme_controller/Makefile])
```

**Comment by timofeymukha** [comment]
The original PR is much expanded based on our discussions. 
* A structural change is that all `time_scheme` class are now stateless, and just implement functions to compute coefficients. They don't store the coefficients and know nothing about what the current time-step is. 
* The time schemes are moved to a separate directory, and each scheme got its own file.
* The logic for selecting the time scheme is now in a separate class called *time_scheme_controller*. It does much of what time schemes did before: stores the coefficients, syncs them with the device, etc.
* The idea is that this new structure will allow us to 
  * Easily add new time schemes, which can be used for any physics.
  * Experiment with new time-stepping logics by creating new controllers. For example, test using a different "start-up" scheme for BDF2/3.
* For now there is only one controller, which implements the time-scheme selection based on the order of the BDF scheme. The logic should follow what we have in Nek5000, and is well-documented in the class. @pschlatt1 could perhaps take a look.
* A type for the Adam-Bashforth scheme has been added, these are used when BDF is of order 1.
* Tests for all the new types.

Note that since Neko used a simpler time scheme selection logic, the PR is in that sense *breaking*.

**Comment by timofeymukha** [comment]
Some functions lack `implicit none`, will add.

File: configure.ac (hunk 1 above)
Comment (njansson) [inline configure.ac:306]:
```suggestion
                 tests/vector/Makefile\
                 tests/scratch_registry/Makefile\
                 tests/fast3d/Makefile\
                 tests/time_scheme/Makefile\
                 tests/time_scheme_controller/Makefile])
```

**Review by njansson — CHANGES_REQUESTED** [review]

**Review by njansson — APPROVED** [review]

**Comment by timofeymukha** [comment]
Ah, right, nvfortran won't work with anything downstream of the stats merge. Waiting for #847  :-P.

**Comment by timofeymukha** [comment]
I'm getting a strange error when trying to compile the branch on LUMI-G.
```
In file included from math/bcknd/device/hip/opr_dudxyz.hip:38:$
In file included from /opt/rocm/hip/include/hip/hip_runtime.h:26:$
In file included from /opt/rocm/hip/include/hip/../../../include/hip/hip_runtime.h:114:$
/opt/rocm/hip/include/hip/library_types.h:25:9: warning: This file is deprecated. Use file from include path /opt/rocm-ver/include/ and prefix with hip [-W#pragma-messages]$
#pragma message("This file is deprecated. Use file from include path /opt/rocm-ver/include/ and prefix with hip")$
        ^$
error: register rewriting failed: cycle in copy bundle$
26 warnings and 1 error generated when compiling for gfx90a.$
make[1]: *** [Makefile:3402: math/bcknd/device/hip/opr_dudxyz.o] Error 1$
make[1]: Leaving directory '/pfs/lustrep4/users/timukha/neko/src'$
make: *** [Makefile:691: all-recursive] Error 1$
```

**Comment by njansson** [comment]
> I'm getting a strange error when trying to compile the branch on LUMI-G.
> 
> ```
> In file included from math/bcknd/device/hip/opr_dudxyz.hip:38:$
> In file included from /opt/rocm/hip/include/hip/hip_runtime.h:26:$
> In file included from /opt/rocm/hip/include/hip/../../../include/hip/hip_runtime.h:114:$
> /opt/rocm/hip/include/hip/library_types.h:25:9: warning: This file is deprecated. Use file from include path /opt/rocm-ver/include/ and prefix with hip [-W#pragma-messages]$
> #pragma message("This file is deprecated. Use file from include path /opt/rocm-ver/include/ and prefix with hip")$
>         ^$
> error: register rewriting failed: cycle in copy bundle$
> 26 warnings and 1 error generated when compiling for gfx90a.$
> make[1]: *** [Makefile:3402: math/bcknd/device/hip/opr_dudxyz.o] Error 1$
> make[1]: Leaving directory '/pfs/lustrep4/users/timukha/neko/src'$
> make: *** [Makefile:691: all-recursive] Error 1$
> ```

Yes there's a known bug in the CCE 15 compiler, which causes an internal compiler error for certain HIP code. The solution is to use `hipcc` instead (configure without specifying `HIPCC` or set it explicitly `HIPCC=hipcc`)

**Comment by timofeymukha** [comment]
Hemi runs on LUMI-G. I think I am comfortable with merging now.

**Review by njansson — APPROVED** [review]

**Pull request merged**